nutanix/provider.go

```diff
@@ -136,26 +167,62 @@ func Provider() *schema.Provider {
 			"nutanix_recovery_plan":           resourceNutanixRecoveryPlan(),
 			"nutanix_service_group":           resourceNutanixServiceGroup(),
 			"nutanix_address_group":           resourceNutanixAddressGroup(),
+			"nutanix_foundation_image_nodes":  resourceFoundationImageNodes(),
+			"nutanix_foundation_ipmi_config":  resourceNutanixFoundationIPMIConfig(),
+			"nutanix_foundation_image":        resourceNutanixFoundationImage(),
 		},
-		ConfigureFunc: providerConfigure,
+		ConfigureContextFunc: providerConfigure,
 	}
 }
 
 // This function used to fetch the configuration params given to our provider which
 // we will use to initialize a dummy client that interacts with API.
-func providerConfigure(d *schema.ResourceData) (interface{}, error) {
+func providerConfigure(ctx context.Context, d *schema.ResourceData) (interface{}, diag.Diagnostics) {
 	log.Printf("[DEBUG] config wait_timeout %d", d.Get("wait_timeout").(int))
 
+	disabledProviders := make([]string, 0)
+	// create warnings for disabled provider services
+	var diags diag.Diagnostics
+	for k, v := range requiredProviderFields {
+		// check if any field is not provided
+		for _, attr := range v {
+			// for string fields
+			if val, ok := d.Get(attr).(string); ok && val == "" {
+				disabledProviders = append(disabledProviders, k)
+				break
+			}
+			// for integer fields
+			if val, ok := d.Get(attr).(float64); ok && int64(val) == 0 {
+				disabledProviders = append(disabledProviders, k)
+				break
+			}
```

Comment:
What about boolean and other types of fields? Are we using any integer fields in the above provider auth config?